depinject/appconfig/config_test.go

```diff
@@ -121,11 +121,122 @@ modules:
 	expectContainerErrorContains(t, opt, "module should have ModuleDescriptor.go_import specified")
 }
 
+func TestComposeGogoTypes(t *testing.T) {
```

Comment:
How is this test different from `TestCompose`?  It looks like a lot of repeated code, maybe just include the *new* test cases which test the new code path, i.e. yamls like
```
    "@type": type.googleapis.com/testpb.TestGogoUnregisteredModule
```

Reply:
yeah its quite same, cause Compose take v1 msg or v2 as input so I separate tests for each case